### PR TITLE
Avoid non-oneliner definition in class and add missing iterators.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: source/opt/basic_block.h [hunk 1]
```diff
@@ -45,16 +46,23 @@ class Function;
...
 class BasicBlock {
  public:
+  using iterator = UptrVectorIterator<Instruction>;
+  using const_iterator = UptrVectorIterator<Instruction, true>;
+
   // Creates a basic block with the given enclosing |function| and starting
   // |label|.
-  BasicBlock(std::unique_ptr<Instruction> label)
-      : function_(nullptr), label_(std::move(label)) {}
+  inline BasicBlock(std::unique_ptr<Instruction> label);
```
Comment (Qining) [inline source/opt/basic_block.h]: Blurbs need update

Comment (antiagainst) [inline source/opt/basic_block.h]: Done.

**Comment by Qining** [comment]
lgtm +1

File: source/opt/function.h [hunk 2]
```diff
@@ -48,17 +48,15 @@ class Function {
   using const_iterator = UptrVectorIterator<BasicBlock, true>;
 
   // Creates a function instance declared by the given instruction |def_inst|.
```
Comment (dneto0) [inline source/opt/function.h]: We know this is has to be an OpFunction instruction, right?  Might as well say so.

**Comment by dneto0** [comment]
Suggested minor improvement to one blurb.
But +2

File: source/opt/function.h (hunk 2 above)
Comment (antiagainst) [inline source/opt/function.h]: Done.

**Pull request merged**